developer/appdev/articles/ui/01-angular.markdown

```diff
@@ -1,61 +1,219 @@
-    web context path, specified in the portlet's `.npmbundlerrc` file. An 
-    example configuration is shown below:
-
-        /o/[web-context]/app/add-entry.component.html
-
-6.  Update the app to use portlet-level styling. Remove the `styleUrl`s from 
-    your component classes, along with the selector properties, and copy the 
-    `app.component.css` file, along with any other component CSS files you have 
-    to `assets/css/` folder.
-
-7.  Import `app.component.css` at the top of the `styles.css` file, along with 
-    any other component CSS files you copied into the `css` folder in the last 
-    step.
-
-        @import "app.component.css";
+1.  Install the Liferay Bundle Generator, using
+    [npm](https://www.npmjs.com/)
+    and
+    [Yeoman](https://yeoman.io/):
+
+        npm install -g yo generator-liferay-bundle
+
+2.  Generate an Angular based portlet bundle project for deploying your app to 
+    your 
+    [@product@ installation](/deployment/docs/installing-product). 
+
+        yo liferay-bundle
+
+    Select `Angular based portlet` and opt for generating sample code. Here's the bundle's structure: 
+
+    -   `my-angular-portlet-bundle`
+        -   `assets` &rarr; CSS, HTML templates, and resources
+            -   `css` &rarr; CSS files
+                -   `styles.css` &rarr; Default CSS file
+            -   `app` &rarr; HTML templates
+                -   `app.component.html` &rarr; Root component template
+        - `features` &rarr; @product@ bundle features
+            -   `localization` &rarr; Resource bundles
+                -   `Language.properties` &rarr; Default language keys
+        -   `src` &rarr; JavaScript an TypeScript files
+            -   `app` &rarr; Application modules and Components
+                -   `app.component.ts` &rarr; Main component
+                -   `app.module.ts` &rarr; Root module
+                -   `dynamic.loader.ts` &rarr; Loads an Angular component dynamically for the portlet to attach to
+            -   `types`
+                -    `LiferayParams.ts` &rarr; Parameters passed by @product@ to the JavaScript module
+            -   `index.ts` &rarr; Main module invoked by the "bootstrap" module to initialize the portlet
+            -   `polyfills.ts` &rarr; Fills in browser JavaScript implementation gaps
+        -   `package.json` &rarr; npm bundle configuration
+        -   `README.md`
+        -   `.npmbuildrc` &rarr; Build configuration
+        -   `.npmbundlerrc` &rarr; Bundler configuration
+        -   `tsconfig.json` &rarr; TypeScript configuration
+
+3.  Copy your app files, matching the types listed below, into your new project.
+
+    | File type | Destination | Comments |
+    | --------- | ----------- | -------- |
+    | HTML | `assets/app/` | Merge with the existing `app.component.html`. Don't add subfolders to `assets/app/`.  |
+    | CSS  | `assets/css/` | Overwrite `styles.css`. |
```

Comment:
Again, here I would reword it to say "Overwrite the `styles.css` file and copy over your your remaining CSS files."